services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java

```diff
@@ -1474,7 +1484,94 @@ public void sendPortUpEvents(final IOFSwitch sw) throws SwitchOperationException
                 }
             }
         }
+    }
+    
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PortConfigurationResponse configurePort(final PortConfigurationRequest request) 
+            throws SwitchOperationException {
+        DatapathId dpId = DatapathId.of(request.getSwitchId());
+        IOFSwitch sw = lookupSwitch(dpId);
+        
+        OFPortDesc ofPortDesc = getPort(sw, request.getPortNo());
+        logger.debug("Switch: {}, port: {}", sw, ofPortDesc);
+        
+        PortStatus newStatus = PortStatus.getPortStatus(request.getStatus());
+        if (newStatus != null) {
+            updatePortStatus(sw, ofPortDesc, newStatus);
+        }
+
+        if (request.getSpeed() > 0) {
+            updatePortSpeed(sw, ofPortDesc, request.getSpeed());
+        }
 
+        return new PortConfigurationResponse(request.getSwitchId(), request.getPortNo(), true);
```

Comment:
What the reason to have this boolean parameter? As I see in case of error you send back ErrorMessage

Reply:
Updated